### PR TITLE
[8.x] Add ability to define default Password Rule

### PR DESCRIPTION
**Pull request by crynobone**
This will make it easier to define a generic default password rules for login, register and password reset and useful for Laravel UI, Breeze, Jetstream and Nova.

### Defining Rules

Typically on the `App\Providers\AuthServiceProvider`
```php
use Illuminate\Validation\Rules\Password;

Password::defaultUsing(function() {
     return Password::min(8)->uncompromised();
});
```

### Using the rules

E.g in Laravel Nova.

```php
use Laravel\Nova\Fields\Password;
use Illuminate\Validation\Rules\Password as PasswordRule;

Password::make('Password', 'password')->rules(PasswordRule::default());
```

Also, you can use:

```php
Password::required(); // return ['required', Password::min(8)->uncompromised()];

Password::sometimes(); // return ['sometimes', Password::min(8)->uncompromised()];
```

Signed-off-by: Mior Muhammad Zaki <crynobone@gmail.com>

### CODE REVIEW

File: src/Illuminate/Validation/Rules/Password.php [hunk 1]
```diff
@@ -85,6 +93,31 @@ public function __construct($min)
...
+
+        static::$defaultCallback = $callback;
+    }
+
+    /**
+     * Get Password's default rules.
+     *
+     * @return array
+     */
+    public static function fromDefault()
```
Comment (driesvints) [inline src/Illuminate/Validation/Rules/Password.php:116]: Maybe just:

```suggestion
    public static function default()
```

**Comment by driesvints** [comment]
Would be cool if you could do:

```php
Password::default()->required();
```

**Comment by crynobone** [comment]
```php
Password::required()
```

I really liking this, `Password::required()` for login and reset password, `Password::sometimes()` for change password.

**Comment by taylorotwell** [comment]
So, if you use `Password::required`, I guess you are supposed to do something like this when building your validation rules?

```php
'name' => ['required', 'string'],
'password' => [...Password::required(), 'foo', 'bar'],
```

**Comment by crynobone** [comment]
@taylorotwell yes, that's the plan with returning it as `array`.

Was tempting to do, but not sure if it's wise

```php
public static function __callStatic($method, $parameters)
{
    return [$method, static::default()];
}
```

**Comment by lloricode** [comment]
Hi, did you just forget `confirmed` rule?

**Comment by crynobone** [comment]
> Hi, did you just forget confirmed rule?

Intentionally leave it out, you can do `[...Password::required(), 'confirmed']` for specific use case.

**Pull request merged**